frontend/app/lib/sprites/tyton.js

```diff
@@ -77,6 +89,30 @@ class Tyton extends Phaser.Sprite {
     this.start_speaking(strings);
   }
 
+  success_task() {
+    this.health += 10;
+    this.power += 10;
+  }
+
+  fail_task() {
+    let damage = this.damage.default;
```

Comment:
if we want to decouple the idea of tyton taking damage from the idea of Tyton failing a task, this is a good idea. however, I would simply write an extra method on Tyton like `take_damage`